src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -615,6 +616,17 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string? ba
                             }
                             continue;
 
+                        case "generatedfilesout":
+                            if (RoslynString.IsNullOrWhiteSpace(value))
+                            {
+                                AddDiagnostic(diagnostics, ErrorCode.ERR_SwitchNeedsString, MessageID.IDS_Text.Localize(), arg);
```

Comment:
>ERR_SwitchNeedsString [](start = 69, length = 21)

I couldn't find `ERR_SwitchNeedsString` in the added sections of the test files. Please add a test, if we don't have one.